src/button/internal.tsx

```diff
@@ -152,6 +160,20 @@ export const InternalButton = React.forwardRef(
       tabIndex: isAnchor && isNotInteractive ? -1 : explicitTabIndex,
     });
 
+    const analyticsMetadata: GeneratedAnalyticsMetadataButtonFragment = disabled
+      ? {}
+      : {
+          action: 'click',
+          detail: { label: '' },
+        };
+    if (__injectAnalyticsComponentMetadata) {
+      analyticsMetadata.component = {
+        name: 'awsui.Button',
+        label: '',
+        properties: { variant, disabled: `${disabled}` },
```

Comment:
Is it intended for disabled to be empty if it is undefined or did you want. it to always have a true/false string?